daisy/instance.go

```diff
@@ -79,13 +80,41 @@ func instanceExists(client daisyCompute.Client, project, zone, instance string)
 	return strIn(instance, instanceCache.exists[project][zone]), nil
 }
 
-// Instance is used to create a GCE instance. Output of serial port 1 will be streamed to the daisy logs directory.
-type Instance struct {
-	compute.Instance
+// MarshalJSON is a workaround to prevent Instance from using compute.Instance's implementation.
+func (i *Instance) MarshalJSON() ([]byte, error) {
+	return json.Marshal(*i)
+}
+
+//InstanceInterface represent abstract Instance across different API stages (Alpha, Beta, API)
+type InstanceInterface interface {
```

Comment:
Per my understanding, InstanceInterface is for avoiding dup code. However, I still see a lot of dup code except for impl of InstanceInterface (for example, populateDisks). If so, why don't we just dup all code for beta and remove InstanceInterface?

Reply:
Code that modifies nested compute children (such as compute disks and their children) would be difficult to extract to the interface. The combination of duplicate code and interfaced code is what I believed will result in least duplicate code with least effort.

Reply:
I see. In my PR I didn't do this because saved dup code is not more than dup code for the interface.

Reply:
It's not only about reducing the total amount of code, but avoiding duplicate code due to maintenance risks: if we're performing complex logic that we later update only for GA, but forget for Beta, it could lead to nasty bugs. However, for some functions, avoiding duplication was very costly (mostly around modifying nested objects) so I went with code duplication approach instead.